src/encoding/armor.js

```diff
@@ -247,6 +247,9 @@ function splitChecksum(text) {
   if (lastEquals >= 0) {
     body = text.slice(0, lastEquals);
     checksum = text.slice(lastEquals + 1);
+    if (checksum.substr(0, 6) === '\n-----') {
```

Comment:
It does work by the way, notice the new tests, plus I field tested it with messages produced by Airmail.